test/framework/src/main/java/org/elasticsearch/bootstrap/BootstrapForTesting.java

```diff
@@ -349,19 +352,17 @@ static Set<URL> parseClassPathWithSymlinks() throws Exception {
     public static void ensureInitialized() {}
 
     /**
-     * Temporarily dsiables security manager for a test.
-     *
-     * <p> This method is only callable by {@link org.elasticsearch.test.ESTestCase}.
+     * Temporarily disables security manager for a test.
      *
      * @return A closeable object which restores the test security manager
      */
     @SuppressWarnings("removal")
     public static Closeable disableTestSecurityManager() {
-        var caller = Thread.currentThread().getStackTrace()[2];
-        if (ESTestCase.class.getName().equals(caller.getClassName()) == false) {
-            throw new SecurityException("Cannot disable test SecurityManager directly. Use @NoSecurityManager to disable on a test suite");
```

Comment:
Note for core/infra: I've relaxed this requirement to enable broader use of disabling security manager for parts of testing. The most important thing is to ensure security manager is not disabled and forgotten. We already return a releasable object, and additional validation ensures we haven't forgotten to re-enable security manager between tests.